nixos/modules/system/boot/networkd.nix

```diff
@@ -1982,6 +1988,62 @@ in
       };
 
       services.resolved.enable = mkDefault true;
+
     })
   ];
+
+  stage1Config = mkMerge [
+    (commonConfig config.boot.initrd)
+
+    { systemd.network.enable = mkDefault config.boot.initrd.network.enable; }
+
+    (mkIf config.boot.initrd.systemd.network.enable {
+
+      systemd.package = pkgs.systemdStage1Network;
+
+      systemd.additionalUpstreamUnits = [
+        "systemd-networkd-wait-online.service"
+        "systemd-networkd.service"
+        "systemd-networkd.socket"
+        "network-online.target"
+        "network-pre.target"
+        "network.target"
+        "nss-lookup.target"
+        "nss-user-lookup.target"
+        "remote-fs-pre.target"
+        "remote-fs.target"
+      ];
+
+      systemd.services.systemd-networkd.wantedBy = [ "initrd.target" ];
+      systemd.services.systemd-networkd.serviceConfig.DynamicUser = true;
+      systemd.targets.network-online.before = [ "initrd.target" ];
+      # initrd is minimal enough that we probably want networking in
+      # the emergency environment.
+      systemd.targets.network-online.wantedBy = [ "initrd.target" "emergency.target" ];
+      systemd.targets.network.wantedBy = ["emergency.target"];
+
+      systemd.contents = {
+        "/etc/systemd/networkd.conf" = renderConfig config.boot.initrd.systemd.network.config;
```

Comment:
I split this between the stage 1 and 2 configs before I had the environment.etc commit. I should probably re-unify them in the commonConfig.